stats/thresholds_test.go

```diff
@@ -25,99 +25,274 @@ import (
 	"testing"
 	"time"
 
-	"github.com/dop251/goja"
 	"github.com/stretchr/testify/assert"
-
 	"go.k6.io/k6/lib/types"
 )
 
 func TestNewThreshold(t *testing.T) {
-	src := `1+1==2`
-	rt := goja.New()
+	t.Parallel()
+
+	src := `rate<0.01`
 	abortOnFail := false
 	gracePeriod := types.NullDurationFrom(2 * time.Second)
-	th, err := newThreshold(src, rt, abortOnFail, gracePeriod)
+	wantParsed := &thresholdExpression{tokenRate, tokenLess, 0.01}
+
+	gotThreshold, err := newThreshold(src, abortOnFail, gracePeriod)
+
 	assert.NoError(t, err)
+	assert.Equal(t, src, gotThreshold.Source)
+	assert.False(t, gotThreshold.LastFailed)
+	assert.Equal(t, abortOnFail, gotThreshold.AbortOnFail)
+	assert.Equal(t, gracePeriod, gotThreshold.AbortGracePeriod)
+	assert.Equal(t, wantParsed, gotThreshold.parsed)
+}
+
+func TestNewThreshold_InvalidThresholdConditionExpression(t *testing.T) {
+	t.Parallel()
+
+	src := "1+1==2"
+	abortOnFail := false
+	gracePeriod := types.NullDurationFrom(2 * time.Second)
+
+	gotThreshold, err := newThreshold(src, abortOnFail, gracePeriod)
+
+	assert.Error(t, err, "instantiating a threshold with an invalid expression should fail")
+	assert.Nil(t, gotThreshold, "instantiating a threshold with an invalid expression should return a nil Threshold")
+}
+
+func TestThreshold_runNoTaint(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		Source           string
+		LastFailed       bool
+		AbortOnFail      bool
+		AbortGracePeriod types.NullDuration
+		parsed           *thresholdExpression
+	}
+	type args struct {
+		sinks map[string]float64
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantOk  bool
+		wantErr bool
+	}{
+		{
+			name:    "valid expression using the > operator over passing threshold",
+			fields:  fields{"rate>0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenGreater, 0.01}},
```

Comment:
Here `fields` has a bunch of ... fields (:stuck_out_tongue: ) three of which are the same in every case. They also coincide with the fields of Threshold.

From the looks of it - the one that is actually used is also `parsed` which IMO can just be part of the top test case struct. 
Also see https://github.com/grafana/k6/pull/2251/files#r756909900 although from my understandign because this is `runNoTaint` that isn't really interesting parameter. Still (maybe an additional test) to check that it doesn't matter will bea good idea :shrug: .

Reply:
For fields, VSCode snippets are at it again, you're right they're quite redundant. Going to address that indeed🙏🏻